src/mailbox.rs

```diff
@@ -1,6 +1,6 @@
 use std::fmt;
 
-#[derive(Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd, Hash)]
```

Comment:
I don't think it's meaningful to implement `Ord` or `PartialOrd` for `Mailbox`.

Reply:
See previous comments.

Reply:
I agree in general (again, since this is an externally expose type). However, I don't think we want to commit to `Mailbox` being `Ord`. That seems like a strong, but ultimately not particularly useful guarantee. And eagerly adopting it would mean that we are committing to continuing to support `Mailbox: Ord` unless we bump the major version number.

Reply:
No you wouldn't commit to anything yet, see [SemVer](http://semver.org/#spec-item-4).

Reply:
Implementing common traits should be done eagerly and we will be done with it. I see where you're coming from that we you have reasons not to implement `Ord` but those should then be made explicit. Can you give a reasonable example where the type could not be `Ord` anymore?

Reply:
I guess I consider this crate stable enough that it should really be 1.0.0, at which point these would be committing. I can absolutely see `Mailbox` gaining fields that are not `Ord`, such as pointers back to the `Client` for further operations on the selected mailbox, which is why I'm hesitant for that to be added. Again though , the other traits seem fine to derive.

I think we'll need @mattnenterprise to look at this.